### PR TITLE
test: add Playwright tests for navigation component

### PR DESCRIPTION
**Pull request by Onibenjo**
## Done

- Adding e2e tests to the navigation flow in 7 spec files under tests/playwright/tests/navigation/:
    - rendering — header classes, logo, dropdown-window structure
    - dropdowns — lazy loading, caching, content verification, link validation, overlay & edge cases
    - search — open/close, submit, reset
    - mobile — menu toggle, escape key, responsive viewports
    - secondary-nav — per-page secondary nav presence, titles, selected states
    - accessibility — keyboard tab order, Enter/Escape, ARIA relationships
    - performance — 404 page fallback

## QA

- Check out that all Playwright CI CD tests pass 
- [List additional steps to QA the new features or prove the bug has been resolved]

## Issue / Card
https://warthogs.atlassian.net/browse/WD-22317

Fixes # 

## Screenshots

[If relevant, please include a screenshot.]

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)

### CODE REVIEW

File: tests/playwright/tests/navigation/dropdowns.spec.ts
Comment (britneywwc) [inline tests/playwright/tests/navigation/dropdowns.spec.ts:1]: Can we use the navigation items from `navigation.yaml` instead of hard-coding them in this file?

File: tests/playwright/tests/navigation/mobile.spec.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    await nav.goto("/");
+  });
+
+  test("menu button is visible with text Menu", async () => {
+    await expect(nav.mobileMenuButton).toBeVisible();
+    await expect(nav.mobileMenuButton).toHaveText("Menu");
+  });
+
+  test("clicking menu button toggles has-menu-open class", async () => {
+    await nav.openMobileMenu();
```
Comment (britneywwc) [inline tests/playwright/tests/navigation/mobile.spec.ts]: This line can be moved to `test.beforeEach` to reduce duplicated calls. The first test will still pass as the button remains visible even when the menu is open

File: tests/playwright/tests/navigation/secondary-nav.spec.ts
Comment (britneywwc) [inline tests/playwright/tests/navigation/secondary-nav.spec.ts:1]: We can also use `secondary-navigation.yaml` here and run the tests through all the items on the file

File: tests/playwright/tests/navigation/rendering.spec.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import { test, expect } from "@playwright/test";
+import { NavigationComponent, NAV_SECTIONS } from "../../helpers/navigation";
+
+test.describe("Navigation rendering", () => {
+  let nav: NavigationComponent;
+
+  test.beforeEach(async ({ page }) => {
+    nav = new NavigationComponent(page);
+    await nav.goto("/");
+  });
```
Comment (britneywwc) [inline tests/playwright/tests/navigation/rendering.spec.ts]: We can use [Playwright Fixtures](https://playwright.dev/docs/test-fixtures) here to reduce the need of duplicated `beforeEach` hooks on similar test environments

File: tests/playwright/tests/navigation/dropdowns.spec.ts
Comment (Onibenjo) [reply tests/playwright/tests/navigation/dropdowns.spec.ts:1]: Yeah, I was trying to prevent adding a yaml reader package but I believe that would provide more coverage for the nav items

File: tests/playwright/tests/navigation/rendering.spec.ts (hunk 2 above)
Comment (Onibenjo) [reply tests/playwright/tests/navigation/rendering.spec.ts]: Nice. Thanks for pointing out

File: tests/playwright/tests/navigation/dropdowns.spec.ts
Comment (britneywwc) [reply tests/playwright/tests/navigation/dropdowns.spec.ts:1]: I think we should use it if possible to catch on bugs when introducing changes to the nav component, let me know what you think

Comment (Onibenjo) [reply tests/playwright/tests/navigation/dropdowns.spec.ts:1]: Yes that makes sense

**Comment by Onibenjo** [comment]
Thanks @britneywwc for the review.

Can you take a look again?

**Review by britneywwc — APPROVED** [review]
LGTM! :shipit:
